controllers/mongodb_users.go

```diff
@@ -62,7 +62,7 @@ func (r ReplicaSetReconciler) updateConnectionStringSecrets(ctx context.Context,
 		pwd := ""
 
 		if user.Database != constants.ExternalDB {
-			secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: secretNamespace}
+			secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: mdb.Namespace}
```

Comment:
This doesn't seem correct. In fact, the `secretNamespace` maps to `mdb.Namespace`.

Reply:
secretNamespace maps to mdb.Namespace unles a connectionStringSecretNamespace is provided, in which case it maps to connectionStringSecretNamespace.

Furthermore, in the ensureUserResources function, the password secret is looked up only in mdb.Namespace.

This mismatch between functions (this function looks up on one of the two and the ensure user resources looks up on only one of them) causes the problem described in the issue.

Reply:
You're absolutely right! Thanks for explaining.